src/activemasternode.cpp

```diff
@@ -154,29 +165,57 @@ void CActiveMasternode::ManageStateInitial()
         return;
     }
 
+    // Default to REMOTE
+    eType = MASTERNODE_REMOTE;
+
+    // Check if wallet funds are available
+    if(!pwalletMain) {
+        strNotCapableReason = "Wallet not available.";
+        LogPrintf("CActiveMasternode::ManageStateInitial -- not capable: %s\n", strNotCapableReason);
+        return;
+    }
+
+    if(pwalletMain->IsLocked()) {
+        strNotCapableReason = "Wallet is locked.";
+        LogPrintf("CActiveMasternode::ManageStateInitial -- not capable: %s\n", strNotCapableReason);
+        return;
+    }
+
+    if(pwalletMain->GetBalance() < 1000*COIN) {
```

Comment:
The idea of this code is to first check if a wallet is available with collateral.  If it is we run in local mode.

If it is not we default to remote mode and try to find our MN in the list.  If we are successful in finding it we run in remote mode, otherwise we go to the "not capable" state.

I don't see the problem with this unless it is because a wallet (but without collateral) is actually needed by a remote masternode.  But you removed that comment and haven't replied to my response there.

If the wallet is required I would just move eType = MASTERNODE_REMOTE to line 183 (ie. after the wallet checks but before the balance check).